### PR TITLE
BUGFIX: MSF-23669 LCM Release fail to sync run-cycle schedules

### PR DESCRIPTION
**Pull request by hung-nguyen-hoang**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1119/ : SUCCESS in 10s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1001/ : FAILURE in 12m 17s

File: lib/gooddata/models/project.rb [hunk 1]
```diff
@@ -517,8 +525,21 @@ def transfer_schedules(from_project, to_project)
             if process_spec.type != :dataload
               executable = schedule_spec[:executable] || (process_spec.type == :ruby ? 'main.rb' : 'main.grf')
             end
+
             params = schedule_parameters(schedule_spec)
-            created_schedule = remote_process.create_schedule(schedule_spec[:cron] || schedule_cache[schedule_spec[:after]], executable, params)
+
+            if trigger_schedule
```
Comment (danh-ung) [inline lib/gooddata/models/project.rb:531]: the variable trigger_schedule is confusing, why trigger_schedule but updating trigger info = null.
We should rename it to: lazy_update_trigger_info

File: lib/gooddata/lcm/actions/synchronize_schedules.rb [hunk 2]
```diff
@@ -61,12 +61,13 @@ def call(params)
             to_projects = info.to
 
             from = development_client.projects(from_project) || fail("Invalid 'from' project specified - '#{from_project}'")
+            allow_cycle_trigger = exist_cycle_trigger(from)
```
Comment (danh-ung) [inline lib/gooddata/lcm/actions/synchronize_schedules.rb:64]: allow_cycle_trigger --> has_cycle_trigger

File: lib/gooddata/models/project.rb [hunk 3]
```diff
@@ -443,7 +443,7 @@ def transfer_user_groups(from_project, to_project)
       # Object to be cloned from. Can be either segment in which case we take
       # the master, client in which case we take its project, string in which
       # case we treat is as an project object or directly project.
-      def transfer_schedules(from_project, to_project)
+      def transfer_schedules(from_project, to_project, allow_cycle_trigger = false)
```
Comment (danh-ung) [inline lib/gooddata/models/project.rb:446]: allow_cycle_trigger --> has_cycle_trigger

File: lib/gooddata/models/project.rb [hunk 4]
```diff
@@ -496,15 +496,23 @@ def transfer_schedules(from_project, to_project)
         end
 
         results = []
+        update_trigger_schedules = []
         loop do # rubocop:disable Metrics/BlockLength
           break if stack.empty?
           state, changed_schedule = stack.shift
+          trigger_schedule = false
```
Comment (danh-ung) [inline lib/gooddata/models/project.rb:503]: trigger_schedule --> lazy_update_trigger_info

File: lib/gooddata/models/project.rb (hunk 1 above)
Comment (hung-nguyen-hoang) [reply lib/gooddata/models/project.rb]: Done update

File: lib/gooddata/lcm/actions/synchronize_schedules.rb (hunk 2 above)
Comment (hung-nguyen-hoang) [reply lib/gooddata/lcm/actions/synchronize_schedules.rb]: Done update

File: lib/gooddata/models/project.rb (hunk 3 above)
Comment (hung-nguyen-hoang) [reply lib/gooddata/models/project.rb]: Done update

File: lib/gooddata/models/project.rb (hunk 4 above)
Comment (hung-nguyen-hoang) [reply lib/gooddata/models/project.rb]: Done update

File: lib/gooddata/lcm/actions/synchronize_schedules.rb (hunk 2 above)
Comment (hung-nguyen-hoang) [reply lib/gooddata/lcm/actions/synchronize_schedules.rb]: Done update

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1120/ : SUCCESS in 8s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1002/ : FAILURE in 10m 41s

**Comment by hung-nguyen-hoang** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1121/ : SUCCESS in 7s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1003/ : FAILURE in 11m 14s

**Comment by hung-nguyen-hoang** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1123/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1005/ : FAILURE in 11m 17s

**Comment by hung-nguyen-hoang** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1124/ : SUCCESS in 9s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1006/ : SUCCESS in 10m 56s

**Comment by hung-nguyen-hoang** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1125/ : SUCCESS in 8s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1007/ : SUCCESS in 15m 44s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1126/ : SUCCESS in 7s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1008/ : SUCCESS in 11m 27s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ruby-gate-PR1865

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-promote-pipeline/177/ : SUCCESS in 13m 01s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-sonarqube-scan-zuul-gate/83/ : SUCCESS in 3m 15s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/625/ : SUCCESS in 21s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-gitops-deploy-pipeline/49/ : SUCCESS in 1m 41s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-fossa-scanning-zuul-el7/168/ : FAILURE in 3m 19s (non-voting)

**Pull request merged**